client/app/components/queries/QueryEditor/index.jsx

```diff
@@ -0,0 +1,181 @@
+              editor.paste(text);
+              $rootScope.$applyAsync();
+              break;
+            }
+            default:
+              break;
+          }
+        }
+      }),
+    [listenForEditorCommand]
+  );
+
+  const handleSelectionChange = useCallback(
+    text => {
+      setSelectedText(text);
+      updateSelectedQuery(text);
+    },
+    [updateSelectedQuery]
+  );
+
+  const formatQuery = useCallback(() => {
+    Query.format(dataSource.syntax || "sql", queryText)
+      .then(updateQuery)
+      .catch(error => notification.error(error));
+  }, [dataSource.syntax, queryText, updateQuery]);
+
+  const toggleAutocomplete = useCallback(state => {
+    setAutocompleteEnabled(state);
+    localOptions.set("liveAutocomplete", state);
+  }, []);
+
+  const modKey = KeyboardShortcuts.modKey;
+
+  return (
+    <section className="editor__wrapper" data-test="QueryEditor">
+      <QueryEditorComponent
+        ref={editorRef}
+        data-executing={queryExecuting ? "true" : null}
+        syntax={dataSource.syntax}
+        value={queryText}
+        schema={schema}
+        autocompleteEnabled={autocompleteAvailable && autocompleteEnabled}
+        onChange={updateQuery}
+        onSelectionChange={handleSelectionChange}
+      />
+
+      <QueryEditorControls
+        addParameterButtonProps={{
+          title: (
+            <React.Fragment>
+              Add New Parameter (<i>{modKey} + P</i>)
+            </React.Fragment>
+          ),
+          onClick: addNewParameter,
+        }}
+        formatButtonProps={{
+          title: (
+            <React.Fragment>
+              Format Query (<i>{modKey} + Shift + F</i>)
+            </React.Fragment>
+          ),
+          onClick: formatQuery,
+        }}
+        saveButtonProps={
+          canEdit && {
+            title: `${modKey} + S`,
+            text: (
+              <React.Fragment>
+                <span className="hidden-xs">Save</span>
+                {isDirty ? "*" : null}
+              </React.Fragment>
+            ),
+            onClick: saveQuery,
+          }
+        }
+        executeButtonProps={{
+          title: `${modKey} + Enter`,
+          disabled: !canExecuteQuery || queryExecuting,
+          onClick: executeQuery,
+          text: <span className="hidden-xs">{selectedText === null ? "Execute" : "Execute Selected"}</span>,
+        }}
+        autocompleteToggleProps={{
+          available: autocompleteAvailable,
+          enabled: autocompleteEnabled,
+          onToggle: toggleAutocomplete,
+        }}
+        dataSourceSelectorProps={{
+          disabled: !isQueryOwner,
+          value: dataSource.id,
+          onChange: updateDataSource,
+          options: map(dataSources, ds => ({ value: ds.id, label: ds.name })),
+        }}
+      />
+    </section>
+  );
+}
+
+QueryEditor.propTypes = {
+  queryText: PropTypes.string.isRequired,
+  schema: Schema, // eslint-disable-line react/no-unused-prop-types
```

Comment:
Is this `eslint-disable-line` still necessary?

Reply:
Probably no :-) BTW - why was that prop unused?

Reply:
:man_shrugging: